### PR TITLE
Prevent starting cleanup task when it's been disabled (5.x)

### PR DESCRIPTION
**Pull request by weralabaj**

### CODE REVIEW

File: src/NServiceBus.RavenDB/Outbox/RavenDBOutboxExtensions.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static EndpointConfiguration SetTimeToKeepDeduplicationData(this Endpoint
             var now = DateTime.UtcNow;
             if (now - timeToKeepDeduplicationData >= now)
             {
-                throw new ArgumentException("Specify a non-negative TimeSpan. The cleanup process removes entries older than the specified time to keep deduplication data, therefore the time span cannot be negative.", "timeToKeepDeduplicationData");
+                throw new ArgumentException("Specify a non-negative TimeSpan. The cleanup process removes entries older than the specified time to keep deduplication data, therefore the time span cannot be negative.", nameof(timeToKeepDeduplicationData));
```
Comment (ramonsmits) [inline src/NServiceBus.RavenDB/Outbox/RavenDBOutboxExtensions.cs]: Shouldn't we mention the API here the customer should use? Shouldn't this be validated in that extension method?

File: src/NServiceBus.RavenDB/Outbox/RavenDBOutboxExtensions.cs [hunk 2]
```diff
@@ -31,10 +31,15 @@ public static EndpointConfiguration SetTimeToKeepDeduplicationData(this Endpoint
...
         /// </summary>
         /// <param name="configuration">The configuration being extended</param>
-        /// <param name="frequencyToRunDeduplicationDataCleanup">The frequency to run the deduplication data cleanup task. By specifying a negative time span (-1) the cleanup task will never run.</param>
+        /// <param name="frequencyToRunDeduplicationDataCleanup">The frequency to run the deduplication data cleanup task. By specifying <code>Timeout.Infinite</code> (-1 milliseconds) the cleanup task will never run.</param>
         /// <returns>The configuration</returns>
         public static EndpointConfiguration SetFrequencyToRunDeduplicationDataCleanup(this EndpointConfiguration configuration, TimeSpan frequencyToRunDeduplicationDataCleanup)
         {
+            if (frequencyToRunDeduplicationDataCleanup <= TimeSpan.Zero && frequencyToRunDeduplicationDataCleanup != System.Threading.Timeout.InfiniteTimeSpan)
+            {
+                throw new ArgumentOutOfRangeException(nameof(frequencyToRunDeduplicationDataCleanup), "Provide a non-negative TimeSpan to specify cleanup task execution frequency or specify -1 milliseconds to disable cleanup.");
```
Comment (weralabaj) [inline src/NServiceBus.RavenDB/Outbox/RavenDBOutboxExtensions.cs]: > specify -1 milliseconds 

better suggest `System.Threading.Timeout.InfiniteTimeSpan`?

**Comment by bording** [comment]
This does not look correct. A patch release should be a `hotfix-5.0.1` branch, and it should be against `master`, not `develop`.

**Comment by ramonsmits** [comment]
Closing in favor of  https://github.com/Particular/NServiceBus.RavenDB/pull/408

**Pull request closed without merging**